### ISSUE TITLE
[Severe bug] All nginx derived containers are breaking

### ISSUE THREAD

**Issue by bhargav3** [webdevops/Dockerfile#174]
Which ever nginx derived image I try to run, I see a bunch of config errors. For example:

````bash
docker run -p 80:80 -it webdevops/hhvm-nginx
````

is throwing out the following error log.

````log
Digest: sha256:acc8e9c6557b63c61485eeebae8989ec58ef69ed4ff4b1dd85f7bb3abeaeca1d
Status: Downloaded newer image for webdevops/hhvm-nginx:latest
-> Executing /opt/docker/provision/entrypoint.d/05-permissions.sh
-> Executing /opt/docker/bin/service.d/supervisor.d//10-init.sh
2017-05-20 01:17:46,588 CRIT Set uid to user 0
2017-05-20 01:17:46,589 WARN Included extra file "/opt/docker/etc/supervisor.d/cron.conf" during parsing
2017-05-20 01:17:46,589 WARN Included extra file "/opt/docker/etc/supervisor.d/dnsmasq.conf" during parsing
2017-05-20 01:17:46,589 WARN Included extra file "/opt/docker/etc/supervisor.d/hhvm.conf" during parsing
2017-05-20 01:17:46,589 WARN Included extra file "/opt/docker/etc/supervisor.d/nginx.conf" during parsing
2017-05-20 01:17:46,589 WARN Included extra file "/opt/docker/etc/supervisor.d/postfix.conf" during parsing
2017-05-20 01:17:46,590 WARN Included extra file "/opt/docker/etc/supervisor.d/ssh.conf" during parsing
2017-05-20 01:17:46,590 WARN Included extra file "/opt/docker/etc/supervisor.d/syslog.conf" during parsing
2017-05-20 01:17:46,599 INFO RPC interface 'supervisor' initialized
2017-05-20 01:17:46,600 INFO supervisord started with pid 1
2017-05-20 01:17:47,602 INFO spawned: 'nginxd' with pid 16
2017-05-20 01:17:47,604 INFO spawned: 'hhvmd' with pid 17
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 01:17:47,613 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 01:17:47,613 INFO success: hhvmd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
-> Executing /opt/docker/bin/service.d/hhvm.d//10-init.sh
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 01:17:47,648 INFO exited: nginxd (exit status 1; not expected)
Cannot open log file: /docker.stdout
2017-05-20 01:17:47,702 INFO spawned: 'nginxd' with pid 22
cap_set_proc failed
2017-05-20 01:17:47,705 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 01:17:47,748 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 01:17:48,755 INFO spawned: 'nginxd' with pid 29
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 01:17:48,765 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 01:17:48,779 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 01:17:49,781 INFO spawned: 'nginxd' with pid 34
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 01:17:49,787 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 01:17:49,801 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 01:17:50,803 INFO spawned: 'nginxd' with pid 39
^C2017-05-20 01:17:50,805 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 01:17:50,805 WARN received SIGINT indicating exit request
2017-05-20 01:17:50,805 INFO waiting for nginxd, hhvmd to die
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 01:17:50,827 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 01:17:50,828 INFO stopped: hhvmd (terminated by SIGTERM)

````

**Comment by mblaschke** [comment]
Seems like a duplicate of https://github.com/webdevops/Dockerfile/issues/156

Can you give us your `host os`, `docker version` and `docker info`?

**Comment by bhargav3** [comment]
Thanks for the hope, I was in the process of rebuilding everything from scratch.
Let me know if this helps in helping me fix the issue.

**host os**

*cat /etc/lsb-release*
````
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.2 LTS"
````
*uname -a*
````
Linux ip-172-31-62-192 3.13.0-48-generic #80-Ubuntu SMP Thu Mar 12 11:16:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
````
**docker version**
````
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
````

**docker info**
````
Containers: 4
Images: 74
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 82
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-48-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 1
Total Memory: 992.5 MiB
Name: ip-172-31-62-192
ID: XFUA:MJDX:WODS:YKAC:HQ5M:RQRW:Y5E4:IF42:LYGX:YHDW:FVQB:CLWY
WARNING: No swap limit support
````

**Comment by mblaschke** [comment]
We're using `/docker.stdout` in our docker containers as link to the stdout/stderr device from the first process (the entrypoint.sh script).

I'm really not sure why this happens. We tested it on Ubuntu, RedHat, SuSE and Debian. Is there any special stuff installed on this Ubuntu host? Is it a VM?

BTW: your docker version is quite old.

**Comment by bhargav3** [comment]
Thanks for the quick response.

This is an AWS EC2 instance and do you recommend I upgrade my docker? Which version have you tested this on?

**Comment by mblaschke** [comment]
We're building and testing (automated testsuite which tests the respons from nginx and other components) before pushing to Docker hub.
We're currently using `Docker version 17.05.0-ce, build 89658be` for building and pushing but started from `1.10.x`. Docker switched to a year-month version numbering so 1.13 was the latest "normal" version number.

The build process can be viewed here: https://build.webdevops.io/

**Comment by mblaschke** [comment]
BTW: if this is a problem with an old docker verison we still should try to fix it. But we need a way to reproduce it

**Comment by bhargav3** [comment]
I upgraded my docker to 

````
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Mon Mar 27 17:10:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Mon Mar 27 17:10:36 2017
 OS/Arch:      linux/amd64
 Experimental: false
````

I ran `docker run -p 80:80 -it webdevops/php-nginx`

and here is the error log:

````
-> Executing /opt/docker/provision/entrypoint.d/05-permissions.sh
-> Executing /opt/docker/bin/service.d/supervisor.d//10-init.sh
2017-05-20 10:06:30,141 CRIT Set uid to user 0
2017-05-20 10:06:30,141 WARN Included extra file "/opt/docker/etc/supervisor.d/cron.conf" during parsing
2017-05-20 10:06:30,141 WARN Included extra file "/opt/docker/etc/supervisor.d/dnsmasq.conf" during parsing
2017-05-20 10:06:30,141 WARN Included extra file "/opt/docker/etc/supervisor.d/nginx.conf" during parsing
2017-05-20 10:06:30,142 WARN Included extra file "/opt/docker/etc/supervisor.d/php-fpm.conf" during parsing
2017-05-20 10:06:30,142 WARN Included extra file "/opt/docker/etc/supervisor.d/postfix.conf" during parsing
2017-05-20 10:06:30,142 WARN Included extra file "/opt/docker/etc/supervisor.d/ssh.conf" during parsing
2017-05-20 10:06:30,142 WARN Included extra file "/opt/docker/etc/supervisor.d/syslog.conf" during parsing
2017-05-20 10:06:30,150 INFO RPC interface 'supervisor' initialized
2017-05-20 10:06:30,150 INFO supervisord started with pid 1
2017-05-20 10:06:31,153 INFO spawned: 'syslogd' with pid 14
2017-05-20 10:06:31,155 INFO spawned: 'nginxd' with pid 15
2017-05-20 10:06:31,156 INFO spawned: 'php-fpmd' with pid 16
2017-05-20 10:06:31,162 INFO spawned: 'crond' with pid 17
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 10:06:31,175 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 10:06:31,175 INFO success: php-fpmd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 10:06:31,175 INFO success: crond entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
-> Executing /opt/docker/bin/service.d/syslog-ng.d//10-init.sh
-> Executing /opt/docker/bin/service.d/cron.d//10-init.sh
-> Executing /opt/docker/bin/service.d/php-fpm.d//10-init.sh
Setting php-fpm user to application
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 10:06:31,233 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 10:06:31,277 INFO spawned: 'nginxd' with pid 38
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 10:06:31,289 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
[20-May-2017 10:06:31] ERROR: failed to open error_log (/docker.stderr): Permission denied (13)
[20-May-2017 10:06:31] ERROR: failed to post process the configuration
[20-May-2017 10:06:31] ERROR: FPM initialization failed
2017-05-20 10:06:31,299 INFO exited: php-fpmd (exit status 78; not expected)
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 10:06:31,316 INFO spawned: 'php-fpmd' with pid 43
2017-05-20 10:06:31,318 INFO success: php-fpmd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 10:06:31,319 INFO exited: nginxd (exit status 1; not expected)
-> Executing /opt/docker/bin/service.d/php-fpm.d//10-init.sh
2017-05-20 10:06:31,330 INFO spawned: 'nginxd' with pid 44
Setting php-fpm user to application
2017-05-20 10:06:31,333 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 10:06:31,377 INFO exited: nginxd (exit status 1; not expected)
2017-05-20 10:06:31,405 INFO spawned: 'nginxd' with pid 52
-> Executing /opt/docker/bin/service.d/nginx.d//10-init.sh
2017-05-20 10:06:31,418 INFO success: nginxd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
[20-May-2017 10:06:31] ERROR: failed to open error_log (/docker.stderr): Permission denied (13)
[20-May-2017 10:06:31] ERROR: failed to post process the configuration
[20-May-2017 10:06:31] ERROR: FPM initialization failed
2017-05-20 10:06:31,428 INFO exited: php-fpmd (exit status 78; not expected)
nginx: [emerg] open() "/docker.stdout" failed (13: Permission denied)
2017-05-20 10:06:31,440 INFO spawned: 'php-fpmd' with pid 57
2017-05-20 10:06:31,442 INFO success: php-fpmd entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-05-20 10:06:31,442 INFO exited: nginxd (exit status 1; not expected)
-> Executing /opt/docker/bin/service.d/php-fpm.d//10-init.sh
2017-05-20 10:06:31,448 INFO spawned: 'nginxd' with pid 58
Setting php-fpm user to application
...
````

Any help is appreciated!

**Comment by bhargav3** [comment]
Do you think if I override log location value, this error will be resolved?

**Comment by bhargav3** [comment]
I have managed to fix this issue by overriding values that has `docker.std*` to `/dev/std*`

Thanks for pointing me in the correct direction.

**Comment by mblaschke** [comment]
But we need a long term solution for this :)
So /dev/std* is working? You're getting full stdout/stderr messages?

**Comment by mblaschke** [comment]
Can you give us the output of `docker run -ti --rm -e LOG_STDOUT=/dev/null webdevops/hhvm-nginx ls -l /proc/1/fd/`?

**Comment by bhargav3** [comment]
I mean logs are secondary since I'm using this image on production and there was huge downtime because I was stuck with this issue.

Here is what the output is for your command

````
Log stdout redirected to /dev/null
total 0
lrwx------ 1 root root 64 May 20 11:19 0 -> /2
lrwx------ 1 root root 64 May 20 11:19 1 -> /2
lrwx------ 1 root root 64 May 20 11:19 2 -> /2
lr-x------ 1 root root 64 May 20 11:19 3 -> /proc/1/fd
````

**Comment by mblaschke** [comment]
You can control /docker.stdout and /docker.stderr with `LOG_STDOUT` and `LOG_STDERR`

So you can set them to /dev/null if you really want to disable the logs.

If you set `/docker.stdout` to `/dev/stdout` it's a redirect to the stdout of the current process so you will lose it.

Will try to find a better solution.

**Comment by bhargav3** [comment]
Okay, great. I set the those env vars to point to `/dev/null` and that seemed to solve my problem. Thanks!

**Comment by bhargav3** [comment]
Another issue I have noticed is that `supervisorctl` is throwing the following error:

````
unix:///var/run/supervisor.sock no such file
````

Any quick fix for this?

**Comment by mblaschke** [comment]
with which command are you starting the container?

**Comment by bhargav3** [comment]
I use docker compose. After starting the containers in the background, I open interactive bash shell using,

`docker exec -it exporter_export_1 bash`

**Comment by mblaschke** [comment]
How old is your webdevops image? please check `docker images|grep webdevops`

**Comment by bhargav3** [comment]
I'm using the tag `debian-8-php7 ` and all the images are 2 days old.

````
webdevops/hhvm-nginx            latest              434a5c4079f1        2 days ago          546 MB
webdevops/php-nginx             debian-8-php7       edfdf1c92c8a        2 days ago          527 MB
webdevops/php-nginx             latest              1aeb7c31be3a        2 days ago          510 MB
webdevops/php-nginx             ubuntu-16.04        1aeb7c31be3a        2 days ago          510 MB
webdevops/php-nginx             ubuntu-14.04        0187568ef179        2 days ago          518 MB
````

**Comment by mblaschke** [comment]
Is the file `/etc/supervisord.conf` still a link to `/opt/docker/etc/supervisor.conf` and have you modified the file inside your container?

the contents of this file should be:

```
[supervisord]
nodaemon=true

[unix_http_server]
file = /.supervisor.sock
chmod = 0700
chown = root:root
username = root
password = {SHA}e982f17bcbe0f724063b708a4f76db211a999304

[supervisorctl]
serverurl = unix:///.supervisor.sock
username = root
password = docker

[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[include]
files = /opt/docker/etc/supervisor.d/*.conf
```

The section of `[supervisorctl]` controls the `supervisorctl` command.

**Comment by bhargav3** [comment]
Yes, I confirm it has the exact same contents and it's still a link to /opt/docker/etc/supervisor.conf

**Comment by mblaschke** [comment]
if i run `docker run -ti --rm webdevops/php-nginx:debian-8-php7 supervisorctl` i get following message:

```
unix:///.supervisor.sock no such file
supervisor>
```

Sure, because supervisord is not running.
Can you post your `docker-compose.yml`?

**Comment by bhargav3** [comment]
Sure, here is my `docker-compose.yml`

````
https-portal:
  image: steveltn/https-portal
  ports:
    - 80:80
    - 443:443
  links:
    - export
  environment:
    - "DOMAINS=example.com -> https://export"
    - "EMAIL=abc@xyz.com"
    - "PRODUCTION=true"
    - "CLIENT_MAX_BODY_SIZE=0"
    - "PROXY_CONNECT_TIMEOUT=3000"
    - "PROXY_SEND_TIMEOUT=3000"
    - "PROXY_READ_TIMEOUT=3000"
    - "KEEPALIVE_TIMEOUT=3000"

export:
  build: .
  ports:
    - "80"
    - "443"
  links:
    - redis
  environment:
    - "LOG_STDOUT=/dev/null"
    - "LOG_STDERR=/dev/null"

redis:
    image: redis
    ports:
      - "6379"
````

**Comment by mblaschke** [comment]
and your `Dockerfile`?

**Comment by bhargav3** [comment]
**Dockerfile**

````
FROM webdevops/php-nginx:debian-8-php7
COPY devops/cron.conf /opt/docker/etc/supervisor.d/cron.conf
RUN rm /etc/nginx/sites-available/default
RUN echo "application ALL = NOPASSWD: ALL" >> /etc/sudoers
COPY devops/nginx.conf /opt/docker/etc/nginx/vhost.conf
COPY devops/queue-worker.conf /opt/docker/etc/supervisor.d/queue-worker.conf
ADD . /application/code
RUN chmod -R 777 /application/code/*
RUN touch /application/code/storage/logs/queue-worker.log
WORKDIR /application/code
ADD devops/ /devops
RUN bash /devops/self-signed-cert-generator.sh
RUN php /usr/local/bin/composer install --no-dev
RUN php artisan migrate --force
RUN echo '* * * * * php /application/code/artisan schedule:run >> /dev/null 2>&1' | crontab -
````

**Comment by mblaschke** [comment]
Please check your `devops/cron.conf` and `devops/queue-worker.con` configuration files if there is something configured for supervisorctl as this would overwrite the base configuration.

**Comment by mblaschke** [comment]
BTW: an easier cron solution will be coming in the next days/weeks. We're planing to replace the system cron with go-crond with better logging support and full environment variable support.

**Comment by bhargav3** [comment]
**queue-worker.conf**

````
[program:queue-worker]
process_name=%(program_name)s_%(process_num)02d
command=/bin/bash -c "export C_FORCE_ROOT=true && php /application/code/artisan queue:work redis --queue=exporter --timeout=600 --sleep=3 --tries=3 --daemon"
autostart=true
autorestart=true
user=root
numprocs=1
redirect_stderr=true
stdout_logfile=/application/code/storage/logs/queue-worker.log


[program:phantomjs-server]
process_name=%(program_name)s_%(process_num)02d
command=/bin/bash -c "export C_FORCE_ROOT=true && /application/code/vendor/jakoch/phantomjs/bin/phantomjs --ssl-protocol=any --ignore-ssl-errors=true /application/code/vendor/jcalderonzumba/gastonjs/src/Client/main.js 8510 1024 768"
autostart=true
autorestart=true
user=root
numprocs=1
redirect_stderr=true
stdout_logfile=/application/code/storage/logs/phantomjs.log
````

**cron.conf**
[group:cron]
programs=crond
priority=25

[program:crond]
command = /usr/sbin/cron -f
process_name=%(program_name)s
startsecs = 0
autostart = true
autorestart = true
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=0
stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=0

**Comment by mblaschke** [comment]
Strange issue. Can you post your `docker-compose --version`?

If the container is running try to get the container hash and execute:

`docker exec -ti CONTAINER_HASH supervisorctl`
Do you get the same issue here?

**Comment by bhargav3** [comment]
*docker-compose version*

````
docker-compose version: 1.5.1
docker-py version: 1.5.0
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
````

*docker exec -it exporter_export_1 supervisorctl*

````
unix:///var/run/supervisor.sock no such file
````

**Comment by mblaschke** [comment]
same with `supervisorctl -c /opt/docker/etc/supervisor.conf`?

can you try to get the result of `grep -Ri -A 10 supervisorctl /etc/supervisor* /opt/docker/etc/supervisor.*`?

**Comment by bhargav3** [comment]
Yup, same with `supervisorctl -c /opt/docker/etc/supervisor.conf` but I noticed supervisor is running

````
root         1  0.0  1.2  55968 13012 ?        Ss   14:26   0:00 /usr/bin/python /usr/bin/supervisord -c /opt/docker/etc/supervisor.conf --logfile /dev/null --pidfile /dev/null --user root
````

`grep -Ri -A 10 supervisorctl /etc/supervisor* /opt/docker/etc/supervisor.*`

````
/etc/supervisor/supervisord.conf:; (supervisorctl/web interface) to work, additional interfaces may be
/etc/supervisor/supervisord.conf-; added by defining them in separate rpcinterface: sections
/etc/supervisor/supervisord.conf-[rpcinterface:supervisor]
/etc/supervisor/supervisord.conf-supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
/etc/supervisor/supervisord.conf-
/etc/supervisor/supervisord.conf:[supervisorctl]
/etc/supervisor/supervisord.conf-serverurl=unix:///var/run/supervisor.sock ; use a unix:// URL  for a unix socket
/etc/supervisor/supervisord.conf-
/etc/supervisor/supervisord.conf-; The [include] section can just contain the "files" setting.  This
/etc/supervisor/supervisord.conf-; setting can list multiple files (separated by whitespace or
/etc/supervisor/supervisord.conf-; newlines).  It can also contain wildcards.  The filenames are
/etc/supervisor/supervisord.conf-; interpreted as relative to this file.  Included files *cannot*
/etc/supervisor/supervisord.conf-; include files themselves.
/etc/supervisor/supervisord.conf-
/etc/supervisor/supervisord.conf-[include]
/etc/supervisor/supervisord.conf-files = /etc/supervisor/conf.d/*.conf
--
/etc/supervisord.conf:[supervisorctl]
/etc/supervisord.conf-serverurl = unix:///.supervisor.sock
/etc/supervisord.conf-username = root
/etc/supervisord.conf-password = docker
/etc/supervisord.conf-
/etc/supervisord.conf-[rpcinterface:supervisor]
/etc/supervisord.conf-supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
/etc/supervisord.conf-
/etc/supervisord.conf-[include]
/etc/supervisord.conf-files = /opt/docker/etc/supervisor.d/*.conf
--
/opt/docker/etc/supervisor.conf:[supervisorctl]
/opt/docker/etc/supervisor.conf-serverurl = unix:///.supervisor.sock
/opt/docker/etc/supervisor.conf-username = root
/opt/docker/etc/supervisor.conf-password = docker
/opt/docker/etc/supervisor.conf-
/opt/docker/etc/supervisor.conf-[rpcinterface:supervisor]
/opt/docker/etc/supervisor.conf-supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
/opt/docker/etc/supervisor.conf-
/opt/docker/etc/supervisor.conf-[include]
/opt/docker/etc/supervisor.conf-files = /opt/docker/etc/supervisor.d/*.conf
````

**Comment by bhargav3** [comment]
See if this helps: https://github.com/Supervisor/supervisor/issues/851

**Comment by mblaschke** [comment]
Oh, the file `/etc/supervisor/supervisord.conf` shouldn't be there but it's also included in our configuration.

It's strage that this will be evalutated because we're not including `/etc/supervisor/*.conf`. Also our tests are fetching the process IDs of the daemons from `supervisorctl` and checks if the are staying up for more then 10 seconds and if there is NO configuration error.
In our tests all is green so there must be somewhere a configuration error.

Anyway we will delete this file when building.

**Comment by bhargav3** [comment]
Okay, thanks! Is there a way I can get `supervisorctl` to work for me so I can see my processes.

**Comment by mblaschke** [comment]
This still doesn't explain why it is working in our tests and not in your container.

**Comment by bhargav3** [comment]
See if this is of any help:

This is the log when I restarted supervisor:

````
Restarting supervisor: 
2017-05-20 14:59:15,740 CRIT Supervisor running as root (no user in config file)
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/queue-worker.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/cron.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/nginx.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/syslog.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/php-fpm.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/ssh.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/postfix.conf" during parsing
2017-05-20 14:59:15,740 WARN Included extra file "/opt/docker/etc/supervisor.d/dnsmasq.conf" during parsing
Error: Another program is already listening on a port that one of our HTTP servers is configured to use.  Shut this program down first before starting supervisord.
For help, use /usr/bin/supervisord -h
````

**Comment by mblaschke** [comment]
`feature/go-logd` could solve this issue by replacing these named pipes with a custom solution (also replaces syslog)

**Comment by mblaschke** [comment]
Now there is https://github.com/webdevops/go-syslogd which provides named sockets and syslog functionality. Will log all messages to stdout

**Comment by mblaschke** [comment]
Can you please try to update your docker and docker-compose version and retry? Your version is quite old and you should try to stay up2date with docker.

**Comment by bhargav3** [comment]
I tried upgrading and seems like this is the latest version available

**docker version**

````
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Mon Mar 27 17:10:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Mon Mar 27 17:10:36 2017
 OS/Arch:      linux/amd64
 Experimental: false
````

**Comment by bhargav3** [comment]
I was able to upgrade and trying out if the bug still exists. Thanks!

**Comment by khanhkid** [comment]
Thank, I just got this problem and solve by upgrade version for `docker compose`
